File: C++/1738. Find Kth Largest XOR Coordinate Value.cpp
class Solution {
public:
    int kthLargestValue(vector<vector<int>>& matrix, int k) {
        int m = matrix.size();
        int n = matrix[0].size();

        vector<int> res;
        vector<vector<int>> arr (m, vector<int>(n, 0));

        arr[0][0] = matrix[0][0];
        res.push_back(arr[0][0]);

        int t;

        for(int i = 1; i < m; ++i) {
            t = arr[i - 1][0] ^ matrix[i][0];
            arr[i][0] = t;
            res.push_back(t);
        }

        for(int j = 1; j < n; ++j) {
            t = arr[0][j - 1] ^ matrix[0][j];
            arr[0][j] = t;
            res.push_back(t);
        }

        for(int i = 1; i < m; ++i) {
            for(int j = 1; j < n; ++j) {
                t = matrix[i][j] ^ arr[i - 1][j] ^ arr[i][j - 1] ^arr[i - 1][j - 1];
                arr[i][j] = t;
                res.push_back(t);
            }
        }

        sort(res.rbegin(), res.rend());
        return res[k - 1];

    }
};

File: Python3/1738. Find Kth Largest XOR Coordinate Value.py
class Solution:
    def kthLargestValue(self, matrix: List[List[int]], k: int) -> int:
        m = len(matrix)
        n = len(matrix[0])
        arr = [[0]*n for _ in range(m)]

        arr[0][0] = matrix[0][0]
        res = [matrix[0][0]]

        for i in range(1, m):
            t = matrix[i][0] ^ arr[i - 1][0]
            arr[i][0] = t
            res.append(t)
        
        for j in range(1, n):
            t = matrix[0][j] ^ arr[0][j - 1]
            arr[0][j] = t
            res.append(t)

        for i in range(1, m):
            for j in range(1, n):
                t = matrix[i][j] ^ arr[i - 1][j] ^ arr[i][j - 1] ^ arr[i - 1][j - 1]
                arr[i][j] = t
                res.append(t)
                
        return sorted(res)[-k]
